### ISSUE TITLE
Button-card intermittently disappears

### ISSUE THREAD

**Issue by frankiboy1** [custom-cards/button-card#381]
I've been using the custom button-card to control the lock to my door for a long time, with great success. However, after updating home assistant to version 0.116.0, the button-card occasionally disappears. This happens both when using a browser on my computer and when using the android app. The button-card addon is already running on the latest version (as far as I can see).

If I reload the page, the button re-appear. The same happens if I select the "Edit dashboard" option. This is a bit more problematic when I use the android app as I haven't found a way to reload the page.

The button is using a fairly simple configuration:

```
entity: light.danalock
name: Ytterdør
show_state: true
state:
  - color: red
    icon: 'mdi:lock'
    value: 'on'
  - color: green
    icon: 'mdi:lock-open'
    value: 'off'
type: 'custom:button-card'
```

Not sure if you can reproduce this. or give me some instructions on how to debug this locally.

**Comment by RomRider** [comment]
I'd try with 0.116.1 as it seems for some people it broke in 0.116.0 but work with the .1
Please let me know if it still doesn't work.

**Comment by frankiboy1** [comment]
I've tested on 0.116.0, 0.116.1 and 0.116.2. They all show the same
behaviour.

I have also downgraded to 0.115.6 and the issue disappeared.

**Comment by dmontes46** [comment]
hi having the same issue after upgrading to 0.116.2

**Comment by elad-bar** [comment]
hi, having the same issue since 0.116.0, upgraded up to .4 - no change

**Comment by RomRider** [comment]
I'll have a look. Any errors in the javascript console?

**Comment by elad-bar** [comment]
no errors in console,
buttons are being displayed just once and then after switch to another view or refresh it disapears,
clearing cache, ctl+f5, shift-f5 not solving the issue,
Switching between different views sometimes make it visible again (nothing in the console indicates something additional loaded or happened that make it work again) - up until next time.

**Comment by RomRider** [comment]
Alright :(
Are all the button-card in the view disappearing? Some of them only? Always the same ones? Any fancy config for those?

**Comment by elad-bar** [comment]
My configuration is extremely fancy but there is no error in console, everything is disappearing when it happens,
Must say that over the last 2 hours it works better... will keep monitor and update tomorrow

**Comment by RomRider** [comment]
For now I can't seem to reproduce in my setup.
Do you use Custom-Header by any chance? Development has stopped and it breaks some things in 0.116.
If so, I'd suggest that you try without CH enabled for a while.

**Comment by frankiboy1** [comment]
The issue I reported has the same behaviour as @elad-bar described. I'm running a very simple setup, which use the built-in cards for everything, except one custom button-card.
So no Custom-Header in my setup.

**Comment by RomRider** [comment]
Alright, I'll continue to investigate. For now I can't reproduce so expect this to take a bit more time, sorry for that.

**Comment by ndriu** [comment]
I have the same issue. Problem persists only on default_view page.

In my configuration I have few custom cards but in almost all cases only standard cards are presented.

**Comment by RomRider** [comment]
Are you saying this affects other custom cards too?

**Comment by ndriu** [comment]
Yes, on main panel I'm using also weather card, custom vertical card, mini graph card and xiaomi vacuum card. 

In mobile app, in most cases only standard cards with entities are presented and sometimes also chart - it is a little bit random. I have two charts and sometimes I get only one, sometimes no one...

So it may be not a problem of button card only.

**Comment by OldNavi** [comment]
Experience the same. Seems it is a relevant to recent changes introduced with 0.116
https://github.com/home-assistant/frontend/issues/7384

https://developers.home-assistant.io/blog/2020/10/02/lazymoreinfo/

**Comment by RomRider** [comment]
I'm aware about this change but I don't use any of the more-info controls.
The only elements I'm using from ha core are:
* ha-card (which is not lazy loaded)
* ha-icon

I don't understand why this happens.

cc: @iantrich, any idea what has changed that could produce this behavior? I believe ha-card is always loaded so that shouldn't be an issue.

**Comment by OldNavi** [comment]
Well, I believe it could be affected, because we can customize icon in a button card. I did a quick repro - if I add custom mdi icon in yaml card config like
`icon: 'mdi-fire'`
I'm getting a described behavior - card intermittently disappears - seems it isn't get even lazy loaded.
if I remove icon customization - everything works as expected.

**Comment by iantrich** [comment]
@RomRider I've got nothing but haven't dug deep on anything yet. I got a similar report on a custom row that uses absolutely nothing from core: https://github.com/iantrich/text-divider-row/issues/12

**Comment by maxwroc** [comment]
@RomRider I found out about this thread because @OldNavi added link to this issue: https://github.com/home-assistant/frontend/issues/7384
I have added there little bit more info - maybe it will be helpful during investigation. 

@iantrich Exactly I have tried removing all the dependencies / core custom elements and the issue still is there. It repros as well with the boilerplate-card from this repo. Can you please reopen the above mentioned issue?

**Comment by RomRider** [comment]
> Well, I believe it could be affected, because we can customize icon in a button card. I did a quick repro - if I add custom mdi icon in yaml card config like
> `icon: 'mdi:fire'`

I use custom icons on all my buttons, I don't believe it's related to that but I might be wrong. If the icon wouldn't load, it wouldn't be displayed but everything else (state, name etc... Basically text) would be.

**Comment by maxwroc** [comment]
FYI fix is on the way: https://github.com/home-assistant/frontend/pull/7407

**Comment by RomRider** [comment]
Great news, thanks for the heads up :)

**Comment by OldNavi** [comment]
> > Well, I believe it could be affected, because we can customize icon in a button card. I did a quick repro - if I add custom mdi icon in yaml card config like
> > `icon: 'mdi:fire'`
> 
> I use custom icons on all my buttons, I don't believe it's related to that but I might be wrong. If the icon wouldn't load, it wouldn't be displayed but everything else (state, name etc... Basically text) would be.

Seems that not an icon was an influencer but rather config change the made card working again - it triggered a condition to update a view as described in home-assistant/frontend#7407

**Comment by teocannata** [comment]
So we should see that bugfix merged into 0.117 version?

**Comment by maxwroc** [comment]
Yes I think that is the plan. AFAIK the fix went into the beta - for which the cut off was today. The 117 is scheduled to be released to public around Oct 28th, if I remember correctly.

**Comment by frankiboy1** [comment]
Tested with 0.117 and it looks like the issue has been fixed.

Thanks!

**Comment by RomRider** [comment]
Seems fixed with 0.117. Closing. Thanks
